src/Hydrodynamics/Hydrodynamics.cpp

```diff
@@ -123,6 +124,67 @@ Hydrodynamics<ndim>::Hydrodynamics(int hydro_forces_aux, int self_gravity_aux, F
 
 
 //=================================================================================================
+//  Hydrodynamics::CreateNewParticle
+/// Create a new hydro particle in the main arrays at the end of the existing particles.
+/// Also triggers the re-creation of ghosts and another tree-build to include new particles.
+/// Needs to be called soon before the load balancing and tree rebuild because it will invalidate
+/// the tree and the ghosts
+//=================================================================================================
+template <int ndim>
+Particle<ndim>& Hydrodynamics<ndim>::CreateNewParticle
+ (const enum ptype _ptype,             ///< [in] ptype of new particle
+  const enum parttype _parttype,       ///< [in] parttype of new particle
+  const int n,                         ///< [in] Current integer time
+  const int level_step,                ///< [in] ..
+  const int level_max,                 ///< [in] Current maximum block timestep level
+  const FLOAT t,                       ///< [in] Current simulation time
+  const FLOAT m,                       ///< [in] Mass of new particle
+  const FLOAT u,                       ///< [in] Specific internal energy of new particle
+  const FLOAT r[ndim],                 ///< [in] Position of new particle
+  const FLOAT v[ndim],                 ///< [in] Velocity of new particle
+  SimulationBase* sim)				   ///< [inout] Simulation object
+{
```

Comment:
Is it really necessary to have all of this info in the functions interface? It seems to me that this function is trying to do two things: 1) create and 2) initialize the particles. Surely we should default construct the particle and leave 2) to the caller. We are returning a reference anyway and leaving initialization to the caller would be more flexible since e.g. dust particles don't need u, but might need other things setting.